Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-details/ContestDetailsPage.tsx

```diff
@@ -0,0 +1,178 @@
+        <>
+            <Link
+              key={`contest-sub-strat-btn-${allowedSubmissionType.id}`}
+              className={styles.allowedLanguageLink}
+              to={`/contests?category=${selectedCategory?.id}&strategy=${allowedSubmissionType.id}`}
+            >
+                {allowedSubmissionType.name}
+            </Link>
+            {' | '}
+        </>
+    ));
+
+    const renderProblemsNames = () => {
+        if (!problems) {
+            return 'The problems for this contest are not public.';
+        }
+        return problems.map((problem) => (
+            <div className={styles.problemNameItem}>
+                {problem.name}
+            </div>
+        ));
+    };
+
+    const renderAdministrationButtons = () => (
+        <div>
+            <Button onClick={() => navigate(`/administration-new/contests/${id}`)}>Edit</Button>
+            <Button className={styles.adminBtn} onClick={() => navigate(`/contests/${id}/compete/results/full`)}>Full Results</Button>
+        </div>
+    );
+
+    const renderContestActionButton = (isCompete: boolean) => {
+        const navigateToUrl = isCompete
+            ? generateContestBtnUrlString(true, id)
+            : generateContestBtnUrlString(false, id);
+        const isDisabled = isCompete
+            ? !canBeCompeted
+            : !canBePracticed;
+        return (
+            <div className={styles.actionBtnWrapper}>
+                <Button
+                  text={isCompete
+                      ? COMPETE_STRING
+                      : PRACTICE_STRING}
+                  state={isDisabled
+                      ? ButtonState.disabled
+                      : ButtonState.enabled}
+                  size={ButtonSize.small}
+                  isCompete={isCompete}
+                  onClick={() => {
+                      if (!isLoggedIn) {
+                          navigate('/login');
+                          return;
+                      }
+                      navigate(navigateToUrl!);
+                  }}
+                />
+                <Link
+                  className={`${isDisabled
+                      ? styles.disabledLink
+                      : ''} ${isCompete
+                      ? styles.greenColor
+                      : ''}`}
+                  to={`/contests/${id}/${isCompete
+                      ? 'compete'
+                      : 'practice'}/results/simple`}
+                >
+                    <i className="fas fa-user" />
+                    <div className={`${styles.underlinedBtnText}`}>
+                        { isCompete
+                            ? 'compete'
+                            : 'practice'}
+                        {' '}
+                        results
+                        {' '}
+                        {isCompete
+                            ? competeParticipantsCount
+                            : practiceParticipantsCount}
+                    </div>
+                </Link>
```

Comment:
Aren't these links displayed in the contest card as well? can we make a reusable component if so?

Reply:
At least the url is good idea to be extracted and reused.